lib/Analysis/IntervalAnalysis.cpp

```diff
@@ -521,14 +593,17 @@ ExpressionValue IntervalDataFlowAnalysis::performBinaryArithmetic(
   ensure(rhs.getExpr(), "cannot perform arithmetic over null rhs smt expr");
 
   auto res = TypeSwitch<Operation *, ExpressionValue>(op)
-                 .Case<AddFeltOp>([&](AddFeltOp _) { return add(smtSolver, lhs, rhs); })
-                 .Case<SubFeltOp>([&](SubFeltOp _) { return sub(smtSolver, lhs, rhs); })
-                 .Case<MulFeltOp>([&](MulFeltOp _) { return mul(smtSolver, lhs, rhs); })
-                 .Case<DivFeltOp>([&](DivFeltOp divOp) { return div(smtSolver, divOp, lhs, rhs); })
-                 .Case<ModFeltOp>([&](ModFeltOp _) { return mod(smtSolver, lhs, rhs); })
-                 .Case<CmpOp>([&](CmpOp cmpOp) {
-    return cmp(smtSolver, cmpOp, lhs, rhs);
-  }).Default([&](Operation *unsupported) {
+                 .Case<AddFeltOp>([&](auto _) { return add(smtSolver, lhs, rhs); })
+                 .Case<SubFeltOp>([&](auto _) { return sub(smtSolver, lhs, rhs); })
+                 .Case<MulFeltOp>([&](auto _) { return mul(smtSolver, lhs, rhs); })
+                 .Case<DivFeltOp>([&](auto divOp) { return div(smtSolver, divOp, lhs, rhs); })
+                 .Case<ModFeltOp>([&](auto _) { return mod(smtSolver, lhs, rhs); })
+                 .Case<CmpOp>([&](auto cmpOp) { return cmp(smtSolver, cmpOp, lhs, rhs); })
+                 .Case<AndBoolOp>([&](auto _) { return boolAnd(smtSolver, lhs, rhs); })
+                 .Case<OrBoolOp>([&](auto _) { return boolOr(smtSolver, lhs, rhs); })
+                 .Case<XorBoolOp>([&](auto _) {
+    return boolXor(smtSolver, lhs, rhs);
+  }).Default([&](auto *unsupported) {
     unsupported->emitWarning(
         "unsupported binary arithmetic operation, defaulting to over-approximated intervals"
```

Comment:
For the compute, I think we should do something with bitwise And since that is super common. Like if you are performing bitwise and with a constant of the form $$2^n-1$$